### ISSUE TITLE
Can't access XMLHttpRequest due to CORS policy when content API call fails.

### ISSUE THREAD

**Issue by bobhy** [TryGhost/Ghost#10932]
### Issue Summary

When call to content API fails (e.g, bad key, bad host name), response does not include an Access-Control-Allow-Origin header.  When it succeeds, response does indeed have this header.
Consequence is that error object returned to caller always has HTTP status 0 making it impossible to show user actionable error messages ("bad server URL" (404), "invalid API key" (401), "Internal server error" (5xx).

WGET and interactive browser do display an error response (JSON object), so this may be an app coding bug.  I'd appreciate gentle guidance if so.

### To Reproduce

1. Invoke content API at public demo server, with deliberately incorrect key. 

```js
const api = new GhostContentAPI({
  url: 'https://demo.ghost.io',
  key: '22444f78447824223cefc4806f',  // last digit replaced by F -- expect 401 error
  version: 'v2' 
});

this.busy = true;

try {
  const ar = await api.tags.browse();
  this.result = jsonFormat(ar, { type: "space" });
} catch (error) {
   this.error = `${error} invoking tags.`;
} finally {
   this.busy = false;
}
```
2. Try to display a reasonable error message.

Expect:
`error` object thrown on failure should include same HTTP status logged in console.  

Observe:
1. HTTP status reported as 0 rather than actual returned status (401 404,5xx), `response` object is undefined.  This is apparently due to CORS policy violation.

```
Failed to load resource: the server responded with a status of 401 () [https://demo.ghost.io/ghost/api/v2/content/tags/?key=22444f78447824223cefc4806f]
Failed to load resource: the server responded with a status of 401 () [https://demo.ghost.io/ghost/api/v2/content/tags/?key=22444f78447824223cefc4806f]
Access to XMLHttpRequest at 'https://demo.ghost.io/ghost/api/v2/content/tags/?key=22444f78447824223cefc4806f' from origin 'http://localhost:8080' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. [http://localhost:8080/]
> error_1
Error: Network Error
  config: Object {url: "https://demo.ghost.io/ghost/api/v2/content/tags/", method: "get", params: 
 Object, …}
  isAxiosError: true
  message: "Network Error"
  > request: XMLHttpRequest {onreadystatechange: , readyState: 4, timeout: 0, …}
    onabort: function handleAbort() { … }
    onerror: function handleError() { … }
    onload: null
    onloadend: null
    onloadstart: null
    onprogress: null
    onreadystatechange: function handleLoad() { … }
    ontimeout: function handleTimeout() { … }
    readyState: 4
    response: ""
    responseText: ""
    responseType: ""
    responseURL: ""
    responseXML: null
    status: 0
    statusText: ""
    timeout: 0
    upload: XMLHttpRequestUpload {onloadstart: null, onprogress: null, onabort: null, …}
    withCredentials: false
    __proto__: XMLHttpRequest {UNSENT: 0, OPENED: 1, HEADERS_RECEIVED: 2, …}
  response: undefined
  stack: "Error: Network Error
    at createError (http://localhost:8080/4.js:2079:15)
    at XMLHttpRequest.handleError (http://localhost:8080/4.js:2341:14)"
  toJSON: function() { … }
  __proto__: Object {constructor: , name: "Error", message: "", …}
```
2. Behaves the same whether I use Ghost content API or `axios` to make the request.

3. `wget` and interactive browser both display a JSON error response.  `wget` might not enforce CORS, but Chrome browser does, yet it is able to get the response. (in this sample, I botched the host URI rather than the key, wget does not save response file if error is 401...)

```
$ wget --no-check-certificate --content-on-error https://demo.ghost.io/ghost/api/v2/content/argle?key=22444f78447824223cefc48062
--2019-07-21 17:41:03--  https://demo.ghost.io/ghost/api/v2/content/argle?key=22444f78447824223cefc48062
Resolving demo.ghost.io (demo.ghost.io)... 104.18.131.189, 104.18.128.189, 104.18.132.189, ...
Connecting to demo.ghost.io (demo.ghost.io)|104.18.131.189|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
Saving to: ‘argle?key=22444f78447824223cefc48062’

argle?key=22444f78447824223cefc48062                                         [ <=>                                                                                                                                                                               ]      73  --.-KB/s    in 0s

2019-07-21 17:41:03 ERROR 404: Not Found.

---------- bob@server3:~ ----------
$ more argle\?key\=22444f78447824223cefc48062
{"errors":[{"message":"Resource not found","errorType":"NotFoundError"}]}
```
`chrome://net-export` logs confirm that the server response really does not include 'Access-Control-Allow-Origin' header when the server returns an API error, and does when the call succeeds. (can attach if requested).  I am no HTTP expert, don't know whether I should be expecting one.  But it sure seems `XMLHttpRequest` is not going to allow access to the error response object without it.

### Technical details:

* Ghost Version: 2.25
* Node Version: 10.15.2
* Browser/OS: Chrome latest / Windows 10 (and others)
* Database: SQLite3

**Comment by allouis** [comment]
Hey @bobhy thanks for the detailed bug report :dancer:

**Comment by allouis** [comment]
@bobhy This is fixed in master now! :tada:

**Comment by eaorak** [comment]
I see that a solution has been provided by I'm having the exact same issue on version 3.3.0. Any ideas?

**Comment by kevinansfield** [comment]
@eaorak if you need any help please use our [forum](https://forum.ghost.org). If it's determined that there's a genuine bug then we can open a new issue with full reproduction steps.
